numba/lowering.py

```diff
@@ -405,6 +405,25 @@ def lower_binop(self, resty, expr):
         return self.context.cast(self.builder, res, signature.return_type,
                                  resty)
 
+
+    def lower_inplace_binop(self, resty, expr):
+        lhs = expr.lhs
+        rhs = expr.rhs
+        lty = self.typeof(lhs.name)
+        rty = self.typeof(rhs.name)
+        lhs = self.loadvar(lhs.name)
+        rhs = self.loadvar(rhs.name)
+        # Get function
+        expr.fn = expr.fn + '='
```

Comment:
> I guess the work around would be change the code producing IR so that fn will be +=.

Yes, exactly.

> Plus we don't need to assign the result of inplace_binop(rhs=$const0.2, lhs=x, fn=+) to another variable.

Actually, we do, because immutable types will return a distinct value which you must assign to the variable being "mutated".